Mathlib/Topology/Support.lean

```diff
@@ -327,17 +354,17 @@ theorem HasCompactMulSupport.mul (hf : HasCompactMulSupport f) (hf' : HasCompact
 
 end Monoid
 
-section DistribMulAction
+section SMulZeroClass
 
-variable [TopologicalSpace α] [MonoidWithZero R] [AddMonoid M] [DistribMulAction R M]
+variable [TopologicalSpace α] [Zero M] [SMulZeroClass R M]
 variable {f : α → R} {f' : α → M} {x : α}
 
 theorem HasCompactSupport.smul_left (hf : HasCompactSupport f') : HasCompactSupport (f • f') := by
   rw [hasCompactSupport_iff_eventuallyEq] at hf ⊢
   exact hf.mono fun x hx => by simp_rw [Pi.smul_apply', hx, Pi.zero_apply, smul_zero]
 #align has_compact_support.smul_left HasCompactSupport.smul_left
 
-end DistribMulAction
+end SMulZeroClass
 
 section SMulWithZero
```

Comment:
In the next section, you can remove `HasCompactSupport.smul_left'`, as it is superseded by the new version of `HasCompactSupport.smul_left`. Well, don't remove it, instead add a deprecated alias to `HasCompactSupport.smul_left`.

Reply:
I use `@[deprecated]` and removed the proof, then I leave `#align`. Is this fine?